routing/routing.go

```diff
@@ -140,12 +142,86 @@ type RouteFilter struct {
 	Index int
 }
 
+// Contains metrics used by LB algorithms
+type LBMetrics struct {
+	inflightRequests int
+	mutex sync.RWMutex
+}
+
+// Increment the number of outstanding requests from the proxy to a given backend.
+func (m *LBMetrics) IncInflightRequest() {
+	m.mutex.Lock()
+	defer m.mutex.Unlock()
+	m.inflightRequests++
+}
+
+// Decrement the number of outstanding requests from the proxy to a given backend.
+func (m *LBMetrics) DecInflightRequest() {
+	m.mutex.Lock()
+	defer m.mutex.Unlock()
+	m.inflightRequests--
+}
+
+// Decrement the number of outstanding requests from the proxy to a given backend.
+func (m *LBMetrics) GetInflightRequests() int{
+	m.mutex.RLock()
+	defer m.mutex.RUnlock()
+	return m.inflightRequests
+}
+
 // LBEndpoint represents the scheme and the host of load balanced
 // backends.
 type LBEndpoint struct {
 	Scheme, Host string
+	Metrics *LBMetrics
+}
+
+type LBEndpointCollection struct {
+	endpoints []LBEndpoint
```

Comment:
We could drop this and use `Itoa`

Reply:
But for this I would need to keep somehow the order of the elements.